include/nanorange/views/reverse.hpp

```diff
@@ -40,19 +40,6 @@ struct reverse_view
         : base_(std::move(r))
     {}
 
-    template <typename R,
-              /* FIXME: This is not to spec, but we get in horrible recursive trouble if it's omitted */
```

Comment:
The same kind of constructor for `common_view` was officially not removed. Libstdc++ has the following comment:

          /* XXX: LWG 3280 didn't remove this constructor, but I think it should?

followed by the entire constructor being commented out.

Reply:
I'll do a full review soon, but just a quick comment -- please refrain from looking at the libstdc++ ranges source code while contributing to NanoRange.

The GPL'd nature of libstdc++ means that we can't use any of its code in NanoRange -- including code that may have been "inspired by" looking at the way they've done things. I know this seems silly and no-one is likely to sue me (or you!) for doing it, but big companies with highly-paid legal departments are very wary of the GPL, and might be reluctant to use NanoRange if they're worried it might have been "tainted" and open them up to exposure. For safety's sake, it's easier just to avoid looking at libstdc++ entirely (or at least the ranges+algorithms part) :).

(The MS STL and libc++ are fine though, as they're under more liberal non-copyleft licences.)

Reply:
I know the legal "problems" of GPL. I did avoid libstdc++ so far and this is the first thing I did take a look at.

If you think this contribution is already "tainted", feel free to close the pull request or otherwise take any actions you deem appropriate.

Reply:
It's all good, thanks for understanding :)